core/src/main/scala/cats/data/EitherT.scala

```diff
@@ -88,11 +88,22 @@ final case class EitherT[F[_], A, B](value: F[Either[A, B]]) {
 
   def map[D](f: B => D)(implicit F: Functor[F]): EitherT[F, A, D] = bimap(identity, f)
 
+  def mapF[N[_], D](fe: F[Either[A, B]] => N[Either[A, D]]): EitherT[N, A, D] = EitherT(fe(value))
+
+  def recoverF[E](pf: PartialFunction[E, Either[A, B]])(implicit ae: ApplicativeError[F, E]): EitherT[F, A, B] =
```

Comment:
This also feels weird to me because you can "recover" to the left side, which is a different semantics from the original `recover`. I suspect that these two different sets of `recover`s and `recoverWith`s should probably best be organized into two different derivations of `MonadError` instances. That way the code organization might be clearer and more importantly, we might be able to test with `MonadError` laws.